File: 389.find-the-difference/solution.py
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        result = {}
        for c in t:
            result[c] = result.get(c, 0) + 1

        for c in s:
            result[c] -= 1

        for c in t:
            if result[c] == 1:
                return c

File: 389.find-the-difference/solution.kt
class Solution {
    fun findTheDifference(s: String, t: String): Char {
        var flag = 0
        val aInt = 'a'.toInt()

        for (i in 0 until s.length) {
            flag = flag xor (1 shl (s[i].toInt() - aInt))
        }
        for (i in 0 until t.length) {
            flag = flag xor (1 shl (t[i].toInt() - aInt))
        }

        var result = aInt
        while (flag != 1) {
            flag = flag shr 1
            result += 1
        }
        return result.toChar()
    }
}
